File: frontend/types/weather-risk.ts
export type WeatherRiskRequest = {
  area?: string;
  latitude?: number;
  longitude?: number;
  days?: number;
};

export type WeatherLocation = {
  name: string;
  latitude: number;
  longitude: number;
  country?: string | null;
  admin1?: string | null;
  location_type?: string | null;
};

export type DiseaseWeatherRisk = {
  disease: string;
  risk_level: "low" | "moderate" | "high";
  score: number;
  rationale: string;
  watch_signs: string[];
};

export type ForecastDaySummary = {
  date: string;
  avg_temperature_c: number;
  max_humidity_percent: number;
  humid_hours: number;
  rain_probability_percent: number;
  precipitation_mm: number;
  disease_pressure: "low" | "moderate" | "high";
};

export type WeatherRiskResponse = {
  location: WeatherLocation;
  source: string;
  forecast_days: number;
  summary: string;
  risks: DiseaseWeatherRisk[];
  daily: ForecastDaySummary[];
  recommendations: string[];
  disclaimer: string;
};

File: backend/app/schemas/weather_risk.py
from pydantic import BaseModel, Field


class WeatherRiskRequest(BaseModel):
    area: str | None = Field(
        default=None,
        description="City, town, region, or address-like location to geocode.",
    )
    latitude: float | None = Field(default=None, ge=-90, le=90)
    longitude: float | None = Field(default=None, ge=-180, le=180)
    days: int = Field(default=7, ge=1, le=14)


class WeatherLocation(BaseModel):
    name: str
    latitude: float
    longitude: float
    country: str | None = None
    admin1: str | None = None
    location_type: str | None = None


class DiseaseWeatherRisk(BaseModel):
    disease: str
    risk_level: str
    score: int
    rationale: str
    watch_signs: list[str]


class ForecastDaySummary(BaseModel):
    date: str
    avg_temperature_c: float
    avg_dew_point_c: float
    max_humidity_percent: float
    humid_hours: int
    rain_probability_percent: float
    precipitation_mm: float
    disease_pressure: str


class WeatherRiskResponse(BaseModel):
    location: WeatherLocation
    source: str
    forecast_days: int
    summary: str
    risks: list[DiseaseWeatherRisk]
    daily: list[ForecastDaySummary]
    recommendations: list[str]
    disclaimer: str
